include/boost/core/bit.hpp

```diff
@@ -47,12 +47,19 @@
 # if __has_builtin(__builtin_bit_cast)
 #  define BOOST_CORE_HAS_BUILTIN_BIT_CAST
 # endif
+# if __has_builtin(__builtin_bswap16)
+#  define BOOST_CORE_HAS_BUILTIN_BSWAP16
+# endif
 #endif
 
-#if defined(BOOST_MSVC) && BOOST_MSVC >= 1926
+#if !defined(BOOST_CORE_HAS_BUILTIN_BIT_CAST) && (defined(BOOST_MSVC) && BOOST_MSVC >= 1926)
```

Comment:
Why is this change needed?

Reply:
The `BOOST_CORE_HAS_BUILTIN_BIT_CAST` macro could be defined above, if MSVC supports `__has_builtin` one day. It's just a sanity check to avoid redefinition of the macro and a potential warning.

It's not essential to the bswap16 change and hence is done in a separate commit. I can remove it if you want.